Syncthing-fork fails to start - tls 1.2 not supported

### Description of the issue
Hello! Fresh install from Google Play on my device fails to start. Original Syncthing 0.10.15 is working fine. Also tried to export settings from working  Syncthing 0.10.15 and import into fork but this didn't help.
In original syncthing `Use HTTPS for GUI` checkbox was empty.
Thank's for any advice!

### Version Information
    App Version: 1.0.0.16
    Android Version: Android 7, latest official firmware, no root
    Device manufacturer: Huawei
    Device model: Nova (CAN-L01\L11)

### Android Log
application log:
```
01-21 23:33:28.217 W/SyncthingNativeCode(11962): 2019/01/21 20:33:28 http: TLS handshake error from 127.0.0.1:39230: remote error: tls: illegal parameter
01-21 23:33:28.221 W/SyncthingNativeCode(11962): 2019/01/21 20:33:28 http: TLS handshake error from 127.0.0.1:39231: tls: client offered an unsupported, maximum protocol version of 302
01-21 23:33:28.227 W/SyncthingNativeCode(11962): 2019/01/21 20:33:28 http: TLS handshake error from 127.0.0.1:39232: tls: client offered an unsupported, maximum protocol version of 301
```
android log:
```
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): ... 16 more
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): Caused by: javax.net.ssl.SSLProtocolException: SSL handshake terminated: ssl=0x7faaf82c00: Failure in SSL library, usually a protocol error
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION (external/boringssl/src/ssl/s3_pkt.c:610 0x7f94b2e100:0x00000001)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:357)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): ... 16 more
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): Caused by: javax.net.ssl.SSLProtocolException: SSL handshake terminated: ssl=0x7faaf82c00: Failure in SSL library, usually a protocol error
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION (external/boringssl/src/ssl/s3_pkt.c:610 0x7f94b2e100:0x00000001)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:357)
01-21 23:37:40.183 W/PollWebGuiAvailableTask(11962): ... 16 more
```

--- a/app/src/main/java/com/nutomic/syncthingandroid/service/Constants.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/service/Constants.java
@@ -180,6 +180,16 @@ public class Constants {
      * to syncthing core v0.14.53+.
      */
     public static Boolean osSupportsTLS12() {
-        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return false;
+        }
+
+        if ("huawei".equalsIgnoreCase(Build.MANUFACTURER)) {
+            if (com.nutomic.syncthingandroid.util.Util.containsIgnoreCase(Build.MODEL, "CAN")) {
+                return false;
+            }
+        }
+
+        return true;
     }
 }

